apps/docs/src/app/platform/component-docs/platform-forms/multi-input/platform-mulit-input-example/platform-mulit-input-example.component.ts

```diff
@@ -0,0 +1,114 @@
+    { firstName: 'Maine', lastName: 'Augusta' },
+    { firstName: 'Maryland', lastName: 'Annapolis' },
+    { firstName: 'Massachusetts', lastName: 'Boston' },
+    { firstName: 'Michigan', lastName: 'Lansing' },
+    { firstName: 'Minnesota', lastName: 'St. Paul' },
+    { firstName: 'Mississippi', lastName: 'Jackson' },
+    { firstName: 'Missouri', lastName: 'Jefferson City' },
+    { firstName: 'Montana', lastName: 'Helena' },
+    { firstName: 'Nebraska', lastName: 'Lincoln' },
+    { firstName: 'Nevada', lastName: 'Carson City' },
+    { firstName: 'New Hampshire', lastName: 'Concord' },
+    { firstName: 'New Jersey', lastName: 'Trenton' },
+    { firstName: 'New Mexico', lastName: 'Santa Fe' },
+    { firstName: 'New York', lastName: 'Albany' },
+    { firstName: 'North Carolina', lastName: 'Raleigh' },
+    { firstName: 'North Dakota', lastName: 'Bismarck' },
+    { firstName: 'Ohio', lastName: 'Columbus' },
+    { firstName: 'Oklahoma', lastName: 'Oklahoma City' },
+    { firstName: 'Oregon', lastName: 'Salem' },
+    { firstName: 'Pennsylvania', lastName: 'Harrisburg' },
+    { firstName: 'Rhode Island', lastName: 'Providence' },
+    { firstName: 'South Carolina', lastName: 'Columbia' },
+    { firstName: 'South Dakota', lastName: 'Pierre' },
+    { firstName: 'Tennessee', lastName: 'Nashville' },
+    { firstName: 'Texas', lastName: 'Austin' },
+    { firstName: 'Utah', lastName: 'Salt Lake City' },
+    { firstName: 'Vermont', lastName: 'Montpelier' },
+    { firstName: 'Virginia', lastName: 'Richmond' },
+    { firstName: 'Washington', lastName: 'Olympia' },
+    { firstName: 'West Virginia', lastName: 'Charleston' },
+    { firstName: 'Wisconsin', lastName: 'Madison' },
+    { firstName: 'Wyoming', lastName: 'Cheyenne' }
+];
+
+export class ListDataProvider extends DataProvider<User> {
+    constructor() {
+        super();
+    }
+    fetch(params: Map<string, string>): Observable<User[]> {
+        let data = list_elements;
+        if (!!params.get('firstName')) {
+            const keyword = params.get('firstName').toLowerCase();
+            data = data.filter((user) => user.firstName.toLowerCase().indexOf(keyword) > -1);
+        }
+        return of(data);
+    }
+}
+
+@Component({
+    selector: 'fdp-platform-multi-input-example',
+    templateUrl: './platform-mulit-input-example.component.html',
+    changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class PlatformMulitInputExampleComponent implements OnInit {
+    _dataSource = new ListDataSource<User>(new ListDataProvider());
+
+    selectedValue: string;
+
+    dropdownValues: User[] | Observable<User[]>;
+
+    public searchString: string;
+
+    @Input()
+    listTemplate: TemplateRef<any>;
+
+    constructor() {
+        this._dataSource.open().subscribe((data) => {
+            this.dropdownValues = data;
+        });
+    }
+    ngOnInit(): void {
+        this._dataSource.match('*');
+    }
+
+    itemSelected($event: any): void {
```

Comment:
same functionality, values, logic seems to be used multiple times. if possible have in one and import where ever required.

Reply:
every example has its own set of data type for its implementation and its search functionality, and data source is added added to all the example just for easability.